src/Microsoft.DotNet.Darc/src/DarcLib/Models/VirtualMonoRepo/SourceManifest.cs

```diff
@@ -43,6 +43,29 @@ public void UpdateVersion(string repository, string uri, string sha, string pack
         }
     }
 
+    public void DeleteSubmodule(SubmoduleRecord record)
+    {
+        var repo = Submodules.FirstOrDefault(r => r.Path == record.Path);
+        if(repo != null)
+        {
+            Submodules.Remove(repo);
+        }
+    }
+
+    public void UpdateSubmodule(SubmoduleRecord record)
```

Comment:
Nit: `record` might be a reserved keyword in C# so a different variable name might be better